powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/fido2/PowerAuthRegistrationProvider.java

```diff
@@ -120,15 +124,14 @@ public RegistrationChallenge findRegistrationChallengeByValue(String application
         final String activationCode = split[0];
 
         // Only allow created activations to be finished
-        final ActivationRecordEntity activationRecordEntity = repositoryCatalogue.getActivationRepository()
-                .findCreatedActivationWithoutLock(applicationId, activationCode, List.of(ActivationStatus.CREATED), currentTimestamp);
+        ActivationRecordEntity activation = activationQueryService.findActivationByCodeWithoutLock(applicationId, activationCode, List.of(ActivationStatus.CREATED), currentTimestamp).orElseThrow(() -> {
+            logger.warn("Activation with activation code: {} could not be obtained. It either does not exist or it already expired.", activationCode);
+            // Rollback is not required, error occurs before writing to database
+            return new Fido2AuthenticationFailedException("Activation could not be found");
```

Comment:
the change of message is intentional?
"Activation failed" to "Activation could not be found"

Reply:
I was trying to provide a more specific message, but we can revert to the original message, no problem.

Reply:
NP, just checking if this was intentional.